contracts/upgradeable_contracts/amb_erc677_to_erc677/HomeStakeTokenFeeManager.sol

```diff
@@ -61,4 +62,25 @@ contract HomeStakeTokenFeeManager is BlockRewardBridge, Ownable {
         uintStorage[FEE] = _fee;
         emit FeeUpdated(_fee);
     }
+
+    /**
+     * @dev Returns the state of the fee manager configuration: whether
+     * it is ready to collect and distribute fee or not.
+     */
+    function isFeeCollectingActivated() public view returns (bool) {
+        return ((address(_blockRewardContract()) != address(0)) && (getFee() > 0));
+    }
+
+    /**
+     * @dev Distributes fee as per the logic of the fee manager.
+     * In this particular case, the amount of fee is passed the block 
+     * reward contract which will mint new tokens and distribute them
+     * among the stakers.
+     * @param _fee amount of tokens to be distributed
+     */
+    function _distributeFee(uint256 _fee) internal {
+        if (address(_blockRewardContract()) != address(0)) {
```

Comment:
Seems that we are already sure, that BlockReward contract is defined, from `isFeeCollectingActivated()` condition: https://github.com/poanetwork/tokenbridge-contracts/blob/0bc587436e6ad8ec196dfedbac7681b6da6d2aac/contracts/upgradeable_contracts/amb_erc677_to_erc677/HomeStakeTokenMediator.sol#L143-L151

Does it make sense to double check it here?

Reply:
@patitonar @k1rill-fedoseev 
Thanks for the question. From my perspective we should leave check that `_blockRewardContract ` is not zero before calling a method from this contract in `_distributeFee`. Just in case if someone will call `_distributeFee` from another place.

But I see that we call `_blockRewardContract` twice and this is not optimal.

What I can do here is to add a comment in the comment before `_distributeFee` that the caller must be sure that the block reward contract is initialized and to remove if-statement with the safety check from this method. What do you think?

Reply:
Yeah, I think it will be fine. Let's add a comment

Reply:
Addressed in 20b3059